Split new-recipe file into recipe/*

mmandlis opened the pull request:
This is mostly just a straightforward copy&paste.
Except the base.js is kind of ugly, any suggestions for improvement?

and btw is it possible to preserve history, if i split one file into multiple files?

dstockwell commented on runtime/recipe/view-connection.js:
```diff
@@ -0,0 +1,103 @@
+
+var assert = require('assert');
+var base = require('./base.js');
+var Connection = require('./base.js').Connection;
```
`base.Connection` since base is already required above?

dstockwell commented:
> and btw is it possible to preserve history, if i split one file into multiple files?

Possibly not on GitHub, but on the command line `git blame` has `-C` and `-M` to better detect lines moved from one file to another.

mmandlis commented:
I removed the base classes, and renamed base.js to util.js.
I also moved recipe-util.js from runtime/ to runtime/recipe.

I'll submit now (because I'm not as confident in merging later, as you are :)), but if you have any more comments about how this change turned out to be, let me know, and I'll happily address them in follow up PR.

Thanks!
